File: samples/tutorials-xamarin/tutorials-2d/tutorial-01/MainActivity.cs
using System;
using Android.App;
using Android.OS;
using Android.Views;
using AndroidX.AppCompat.App;
using SciChart.Charting.Model.DataSeries;
using SciChart.Charting.Visuals;
using SciChart.Charting.Visuals.Axes;
using SciChart.Charting.Visuals.PointMarkers;
using SciChart.Charting.Visuals.RenderableSeries;
using SciChart.Drawing.Common;

namespace Tutorial_2d_01
{
    [Activity(Label = "@string/app_name", Theme = "@style/AppTheme.NoActionBar", MainLauncher = true)]
    public class MainActivity : AppCompatActivity
    {
        protected override void OnCreate(Bundle savedInstanceState)
        {
            base.OnCreate(savedInstanceState);
            Xamarin.Essentials.Platform.Init(this, savedInstanceState);

            SciChartSurface.SetRuntimeLicenseKey("");
            SetContentView(Resource.Layout.activity_main);

            var surface = new SciChartSurface(this);
            var chartLayout = FindViewById<ViewGroup>(Resource.Id.chart_layout);
            chartLayout.AddView(surface);

            // <AddingAxesToTheSciChartSurface>
            var xAxis = new NumericAxis(this);
            var yAxis = new NumericAxis(this);

            using (surface.SuspendUpdates())
            {
                surface.XAxes.Add(xAxis);
                surface.YAxes.Add(yAxis);
            }
            // </AddingAxesToTheSciChartSurface>

            // <DeclareDataSeries>
            var lineDataSeries = new XyDataSeries<int, double>();
            var scatterDataSeries = new XyDataSeries<int, double>();

            for (int i = 0; i < 999; i++)
            {
                lineDataSeries.Append(i, Math.Sin(i * 0.1));
                scatterDataSeries.Append(i, Math.Cos(i * 0.1));
            }
            // </DeclareDataSeries>

            // <CreateRenderableSeries>
            var lineSeries = new FastLineRenderableSeries() { DataSeries = lineDataSeries };
            var scatterSeries = new XyScatterRenderableSeries()
            {
                DataSeries = scatterDataSeries,
                PointMarker = new EllipsePointMarker()
                {
                    Width = 10,
                    Height = 10,
                    FillStyle = new SolidBrushStyle(0xFF32CD32)
                }
            };
            // </CreateRenderableSeries>

            // <AddRenderableSeries>
            using (surface.SuspendUpdates())
            {
                surface.RenderableSeries.Add(lineSeries);
                surface.RenderableSeries.Add(scatterSeries);
            }
            // </AddRenderableSeries>
        }
    }
}

File: samples/first-app/JetpackCompose/src/main/java/com/example/firstscichartapp/MainActivity.kt
package com.example.firstscichartapp

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.compose.runtime.Composable
import androidx.compose.ui.viewinterop.AndroidView
import com.example.firstscichartapp.databinding.ActivityMainBinding
import com.scichart.charting.model.dataSeries.XyDataSeries
import com.scichart.charting.visuals.SciChartSurface
import com.scichart.charting.visuals.axes.IAxis
import com.scichart.charting.visuals.axes.NumericAxis
import com.scichart.charting.visuals.renderableSeries.FastLineRenderableSeries
import com.scichart.core.framework.UpdateSuspender
import com.scichart.core.model.DoubleValues
import kotlin.math.sin

// <FinalExampleCodeUsingCompose>
class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.composeView.setContent {
            SciChartView()
        }
    }
}

@Composable
fun SciChartView() {
    AndroidView(factory = { context ->
        SciChartSurface(context)
    }, update = { surface ->
        val xAxis: IAxis = NumericAxis(surface.context)
        val yAxis: IAxis = NumericAxis(surface.context)

        UpdateSuspender.using(surface) {
            surface.xAxes.add(xAxis)
            surface.yAxes.add(yAxis)
        }

        val count = 1000
        val xValues = DoubleValues(count)
        val yValues = DoubleValues(count)

        for (i in 0 until count) {
            val x = 10.0 * i.toDouble() / count.toDouble()
            val y = sin(2 * x)
            xValues.add(x)
            yValues.add(y)
        }

        val dataSeries =
            XyDataSeries(Double::class.javaObjectType, Double::class.javaObjectType).apply {
                append(xValues, yValues)
            }

        val renderableSeries = FastLineRenderableSeries().apply {
            this.dataSeries = dataSeries
        }

        surface.renderableSeries.add(renderableSeries)
    })
}
// </FinalExampleCodeUsingCompose>


/*
// <AddingChartSurfaceUsingCompose>
AndroidView(factory = { ctx ->
    SciChartSurface(ctx).apply {
        // configure your surface with axes, dataSeries, renderableSeries, etc.
    }
}, update = {
    // update your surface if needed
})
// </AddingChartSurfaceUsingCompose>
 */